File: Session18/14. Longest Common Prefix.py

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:

        str1, str2 = min(strs), max(strs)
        i = 0
        while i < len(str1):
            if str1[i] != str2[i]:
                break
            i +=1

        return str1[:i]

File: Session18/14. Longest Common Prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        
        Arrays.sort(strs);
        
        int n=strs.length;
        
        int maxiter=Math.min(strs[0].length(),strs[n-1].length());
        int i=0;
            
        while(i<maxiter)
        {
            if(strs[0].charAt(i)==strs[n-1].charAt(i))
            {
                i++;
            }
            else
            {
                break;
            }
        }
        
        return strs[0].substring(0,i);
    }
}